synapse/handlers/account_validity.py

```diff
@@ -261,4 +268,28 @@ def renew_account_for_user(self, user_id, expiration_ts=None, email_sent=False):
             email_sent=email_sent,
         )
 
+        # Check if renewed users should be reintroduced to the user directory
+        if self.hs.config.show_users_in_user_directory:
+            # Show the user in the directory again by setting them to active
+            yield self.profile_handler.set_active(UserID.from_string(user_id), True, True)
+
         defer.returnValue(expiration_ts)
+
+    @defer.inlineCallbacks
+    def _mark_expired_users_as_inactive(self):
+        """Iterate over expired users. Mark them as inactive in order to hide them from the
+        user directory.
+
+        Returns:
+            Deferred
```

Comment:
This looks like to be a lie because `set_active` doesn't return anything.

Reply:
Any method with `@defer.inlineCallbacks` immediately returns a Deferred. Whether we actually want to call this out in the docstring is probably a stylistic choice though...

Reply:
Apparently we want it.

Reply:
Apparently some IDEs use that info to figure out whether you need to yield, so let's keep it.